### PR TITLE
Azure: Remove vnet creation dependency from provisioner

### PR DESCRIPTION
**Pull request by kartikjoshi21**
Use default vnet and fetch defaut vnet subnet id in azure provisioner

Fixes: #1221

### CODE REVIEW

File: test/provisioner/provision_azure.go [hunk 1]
```diff
@@ -326,16 +269,51 @@ func (p *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.
...
 
-	aksOIDCIssuer := *cluster.Properties.OidcIssuerProfile.IssuerURL
+	aksOIDCIssuer := *clusterpre.Properties.OidcIssuerProfile.IssuerURL
 	if err := createFederatedIdentityCredential(aksOIDCIssuer); err != nil {
 		return fmt.Errorf("creating federated identity credential: %w", err)
 	}
 
+	cluster, err := AzureProps.ManagedAksClient.Get(context.Background(), AzureProps.ResourceGroupName, AzureProps.ClusterName, nil)
+	if err != nil {
+		return fmt.Errorf("Failed to get AKS cluster: %q details: %w", AzureProps.ClusterName, err)
```
Comment (surajssd) [inline test/provisioner/provision_azure.go:284]:
```suggestion
		return fmt.Errorf("getting AKS cluster object %q: %w", AzureProps.ClusterName, err)
```

File: test/provisioner/provision_azure.go [hunk 2]
```diff
@@ -326,16 +269,51 @@ func (p *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.
...
+	// Fetch aks-rg details
+	aks_rg := *cluster.Properties.NodeResourceGroup
+
+	// Fetch default vnet name
+	vnetName := ""
+	pager := AzureProps.ManagedVnetClient.NewListPager(aks_rg, nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
+		if err != nil {
+			return fmt.Errorf("Failed to get vnets associated with aks cluster: %q details: %w", AzureProps.ClusterName, err)
```
Comment (surajssd) [inline test/provisioner/provision_azure.go:296]:
```suggestion
			return fmt.Errorf("getting VNETs of AKS cluster: %q: %w", AzureProps.ClusterName, err)
```

File: test/provisioner/provision_azure.go [hunk 3]
```diff
@@ -326,16 +269,51 @@ func (p *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.
...
+		}
+		for _, v := range nextResult.Value {
+			vnetName = *v.Name
+		}
+	}
+
+	virtualNetwork, err := AzureProps.ManagedVnetClient.Get(ctx, aks_rg, vnetName, nil)
+	if err != nil {
+		return fmt.Errorf("failed to fetch vnet: %q: %v", vnetName, err)
```
Comment (surajssd) [inline test/provisioner/provision_azure.go:305]:
```suggestion
		return fmt.Errorf("fetching VNET: %q: %w", vnetName, err)
```

File: test/provisioner/provision_azure.go [hunk 4]
```diff
@@ -326,16 +269,51 @@ func (p *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.
...
+	}
+
+	SubnetsPtr := &virtualNetwork.Properties.Subnets
+	if SubnetsPtr == nil || len(*SubnetsPtr) == 0 {
+		return fmt.Errorf("No subnets found in the specified vnet: %q: %v", vnetName, err)
```
Comment (surajssd) [inline test/provisioner/provision_azure.go:310]:
```suggestion
		return fmt.Errorf("no subnet found in the specified VNET: %q", vnetName)
```

File: test/provisioner/provision_azure.go [hunk 5]
```diff
@@ -326,16 +269,51 @@ func (p *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.
...
 		return fmt.Errorf("creating federated identity credential: %w", err)
 	}
 
+	cluster, err := AzureProps.ManagedAksClient.Get(context.Background(), AzureProps.ResourceGroupName, AzureProps.ClusterName, nil)
+	if err != nil {
+		return fmt.Errorf("Failed to get AKS cluster: %q details: %w", AzureProps.ClusterName, err)
+	}
+
+	// Fetch aks-rg details
+	aks_rg := *cluster.Properties.NodeResourceGroup
```
Comment (surajssd) [inline test/provisioner/provision_azure.go:288]: Can't we get this from the existing AKS object? Or we can always generate this name, thus reducing the number of calls that happen to Azure API. 

https://learn.microsoft.com/en-us/azure/aks/faq#why-are-two-resource-groups-created-with-aks

**Review by surajssd — CHANGES_REQUESTED** [review]

**Review by surajssd — APPROVED** [review]

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @kartikjoshi21

**Review by mkulke — APPROVED** [review]
lgtm

**Pull request merged**